test/Core.Test/Services/OrganizationServiceTests.cs

```diff
@@ -194,6 +194,19 @@ await sutProvider.GetDependency<IReferenceEventService>().Received(1)
         await sutProvider.GetDependency<IOrganizationRepository>().Received(1).ReplaceAsync(organization);
     }
 
+    [Theory]
+    [FreeOrganizationUpgradeCustomize, BitAutoData]
+    public async Task UpgradePlan_WithSecretsManager_Passes(Organization organization, OrganizationUpgrade upgrade,
+        SutProvider<OrganizationService> sutProvider)
+    {
+        upgrade.UseSecretsManager = true;
+        upgrade.AdditionalSmSeats = 2;
+        upgrade.AdditionalServiceAccounts = 2;
+        sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(organization.Id).Returns(organization);
+        await sutProvider.Sut.UpgradePlanAsync(organization.Id, upgrade);
+        await sutProvider.GetDependency<IOrganizationRepository>().Received(1).ReplaceAsync(organization);
```

Comment:
Please ignore me if you're still working on this. But at the moment, this doesn't test anything. You'll need to use [argument matchers](https://nsubstitute.github.io/help/argument-matchers/) to inspect the `organization` object that `OrganizationRepository` receives and make sure it has the correct properties.

You should also assert calls to `IPaymentService` to make sure it calls the stripe service properly.

I also recommend setting seats + service accounts to different values, to make sure they don't get mixed up.